src/pages/FrontPage.js

```diff
@@ -1,57 +1,37 @@
 import React, {PropTypes} from 'react';
-import withStyles from 'isomorphic-style-loader/lib/withStyles';
 import {connect} from 'react-redux';
 import Grid from 'react-bootstrap/lib/Grid';
 import Row from 'react-bootstrap/lib/Row';
 import Col from 'react-bootstrap/lib/Col';
-
-import styles from './FrontPage.scss';
 import Filter from '../components/FrontPage/Filter';
 import {CoursesContainer} from '../components/FrontPage/Courses';
 import TeacherInfobox from '../components/FrontPage/TeacherInfobox';
-import Button from 'react-bootstrap/lib/Button';
-import LinkContainer from 'react-router-bootstrap/lib/LinkContainer';
-
-import {setModeTeacher} from '../action_creators';
-import {getTranslator} from '../selectors/translate';
+import WelcomeBox from '../components/FrontPage/WelcomeBox';
 
 export const  FrontPage = React.createClass({
 
-  contextTypes: {
-    router: React.PropTypes.object
-  },
-
   render() {
+    const {isStudentMode} = this.props;
     return (
       <Grid fluid={true}>
 
-        {/* Buttons */}
-        {this.props.isStudentMode
+        {/*WelcomeBox*/}
+        {isStudentMode
```

Comment:
Veldig bra med opprydningen i FrontPage. Det kan gjøres enda litt bedre. På frontpage gjør du en sjekk på isStudentMode og renderer WelcomeBox dersom den er true. Du gjør den samme sjekken for TeacherInfoBox, men dette skjer inni TeacherInfoBox. Det er litt forvirrende.

Jeg forslår at enten så gjør begge komponentene det selv, eller så gjør FrontPage det for begge. Jeg foreslår det siste. Altså
```
{isStudentMode ? <WelcomeBox /> : <TeacherInfobox />}
```
Du trenger da ikke sende inn isStudentMode (du vet jo hva den er), og du kan også flytte inn `<Row>` (dersom du mener det er riktigere å beholde `<Row>` i FrontPage, så gjør du det).